java_src/foam/nanos/auth/AuthOAO.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.oao.ProxyOAO;
+
+import javax.security.auth.AuthPermission;
+import java.util.Map;
+
+public class AuthOAO
+  extends ProxyOAO
+{
+  private final String rootPermission;
```

Comment:
Make private variables protected instead and name them with a trailing _. So:
protected final String rootPermission_;

Protected is just as safe from our point of view but easier to debug.